### PR TITLE
Set category for new meter.

### PR DESCRIPTION
**Pull request by shahardo**
#386

### CODE REVIEW

File: negawatt/modules/custom/negawatt_meter/negawatt_meter.module [hunk 1]
```diff
@@ -206,4 +206,71 @@ function negawatt_meter_field_attach_create_bundle($entity_type, $bundle) {
...
+      'settings' => array(
+        'max_length' => 255,
+      ),
+      'translatable' => 0,
+      'type' => 'text',
+    );
+    field_create_field($field);
+  }
+
+  $instance = array(
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_meter/negawatt_meter.module]: `if (!field_info_instance()`

File: negawatt/modules/custom/negawatt_restful/plugins/restful/node/meters/iec_meters/1.0/NegawattIecMeterResource.class.php [hunk 2]
```diff
@@ -48,6 +48,49 @@ public function createEntity() {
       return parent::updateEntity($id);
     }
     // New node.
-    return parent::createEntity();
+    $return = parent::createEntity();
+
+
+    // Set meter category.
+
+    $new_meter_id = $return[0]['id'];
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/plugins/restful/node/meters/iec_meters/1.0/NegawattIecMeterResource.class.php]: move to `node_presave() if empty($node->nid)`

File: negawatt/modules/custom/negawatt_restful/plugins/restful/node/meters/iec_meters/1.0/NegawattIecMeterResource.class.php [hunk 3]
```diff
@@ -48,6 +48,49 @@ public function createEntity() {
...
+
+    // Get vocabulary according to meter account (organic-group).
+    $account = $wrapper->{OG_AUDIENCE_FIELD}->value();
+    $group_id = $account->nid;
+    $arr_vocabularies = og_vocab_relation_get_by_group('node', $group_id);
+    $vocabulary_id = $arr_vocabularies[0]->vid;
+    $vocabulary = taxonomy_vocabulary_load($vocabulary_id);
+
+    // Try match meter description with any of the categories' match-strings
+    $category = 'אחר';
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/plugins/restful/node/meters/iec_meters/1.0/NegawattIecMeterResource.class.php]: use ``variable_get()``

File: negawatt/modules/custom/negawatt_restful/tests/NegawattRestfulTestCase.test [hunk 4]
```diff
@@ -28,9 +28,79 @@ class NegawattRestfulTestCase extends NegawattWebTestCase {
...
+    $account_node_info->comment  = 0;
+
+    $account_node_info->og_description   = t("Meter Account");
+    $account_node_info->og_register  = 0;
+    $account_node_info->og_directory = 0;
+    $account_node_info->og_private   = 1;
+    $account_node_info->og_selective = 3;
+
+    $account_node1 = node_submit($account_node_info);
+    node_save($account_node1);
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/tests/NegawattRestfulTestCase.test]: ``php
$settings = array(
      'type' => 'account',
      OG_GROUP_FIELD => array('und' => array('0' => array('value' => 1))),
    );

  $node = $this->drupalCreateNode($settings);

```

Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/tests/NegawattRestfulTestCase.test]: move to helper function.

File: negawatt/modules/custom/negawatt_restful/tests/NegawattRestfulTestCase.test [hunk 5]
```diff
@@ -39,15 +109,21 @@ class NegawattRestfulTestCase extends NegawattWebTestCase {
...
       'field_place_address' => array('und' => array('0' => array('value' => 'לכיש 25'))),
       'field_place_locality' => array('und' => array('0' => array('value' => 'קרית גת'))),
+      OG_AUDIENCE_FIELD => array('und' => array('0' => array('target_id' => 1))),
     );
 
     $meter_node1 = $this->drupalCreateNode($settings);
     $wrapper = entity_metadata_wrapper('node', $meter_node1);
     $wrapper->{OG_VOCAB_FIELD}->set(array(7));
+    // Add node to account-1
+    og_group($account_node1->nid, array('entity type' => 'node', 'entity' => $meter_node1,'state' => OG_STATE_ACTIVE));
```
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/tests/NegawattRestfulTestCase.test]: no need for test

File: negawatt/modules/custom/negawatt_restful/tests/NegawattRestfulTestCase.test (hunk 4 above)
Comment (amitaibu) [inline negawatt/modules/custom/negawatt_restful/tests/NegawattRestfulTestCase.test]:
```php
$values = array(
  'vid' => $vocab->vid,
  'name' => 'term name',
);

$term = entity_create('taxonomy_term', $values);
$wrapper ...
$wrapper->save();
```

**Comment by ceoaliongroo** [comment]
@shahardo is this PR wip?

**Comment by shahardo** [comment]
Sort of: functionality works, but I have to add tests
בתאריך 28 באפר' 2015 13:36,‏ "Carlos Mantilla" <notifications@github.com>
כתב:

**Comment by ceoaliongroo** [comment]
ok, i added `WIP`

File: negawatt/modules/custom/iec/iec.module [hunk 6]
```diff
@@ -7,10 +7,63 @@
 include_once 'iec.features.inc';
 
 /**
+ * Set meter category by trying to match description with categories'
+ * match-strings.
+ *
+ * @param $wrapper
+ *  Wrapper for the new node.
+ */
+function set_meter_category($wrapper) {
```
Comment (amitaibu) [inline negawatt/modules/custom/iec/iec.module]: ``iec_set_...``

File: negawatt/modules/custom/iec/iec.module [hunk 7]
```diff
@@ -7,10 +7,63 @@
...
+
+  // Get vocabulary according to meter account (organic-group).
+  $account = $wrapper->{OG_AUDIENCE_FIELD}->value();
+  $group_id = $account->nid;
+  $arr_vocabularies = og_vocab_relation_get_by_group('node', $group_id);
+  $vocabulary_id = $arr_vocabularies[0]->vid;
+  $vocabulary = taxonomy_vocabulary_load($vocabulary_id);
+
+  // Try match meter description with any of the categories' match-strings
+  $category = variable_get('default_iec_meter_category', 'אחר');
```
Comment (amitaibu) [inline negawatt/modules/custom/iec/iec.module]: => `iec_default_meter_category`

**Pull request merged**